src/expr/impl/src/scalar/cast.rs

```diff
@@ -177,40 +177,68 @@ pub fn str_to_bytea(elem: &str) -> Result<Box<[u8]>> {
 
 // TODO(nanderstabel): optimize for multidimensional List. Depth can be given as a parameter to this
 // function.
+// FIXME(runji): handle escape characters in double quoted strings.
```

Comment:
What about decouple this fix to its own PR? Here you can input `text[]` with the following syntax:
```
select jsonb_object(array['a', '1', 'b', 'def', 'c', '3.5']);
```